### DISCUSSION TITLE
Option to auto-reload a SecretStr setting

### DISCUSSION THREAD

**Discussion by timofurrer** [pydantic/pydantic#3854] [Feature Request]
It would be nice to have an option to auto-reload the value of a `SecretStr` from source, e.g. from the file it was initially loaded from. Whereas, *auto-reload* could also mean that it's just read from the original source every time `get_secret_value()` is called.

The use case for that is that the secret in the file is rotate by some other mechanism, e.g. a vault-agent.

**Comment by donvargax** [comment]
I'd like to work on this if there's interest, I need to support the use case explained.

**Comment by ASR-ASU** [comment]
That's something that will be very helpful to avoid, for example, killing a pod each time vault rotates the database keys.
